File: CodeWars/5kyu/JavaScript/004-human-readable-time.js
// https://www.codewars.com/kata/52685f7382004e774f0001f7
// Write a function, which takes a non-negative integer (seconds) as input and returns the time in a human-readable format (HH:MM:SS)
// HH = hours, padded to 2 digits, range: 00 - 99
// MM = minutes, padded to 2 digits, range: 00 - 59
// SS = seconds, padded to 2 digits, range: 00 - 59
// The maximum time never exceeds 359999 (99:59:59)

function humanReadable(seconds) {
    //get hours, mins, second from the input
    let h = parseInt(seconds / 3600);
    seconds = seconds - h * 3600;
    let m = parseInt(seconds / 60);
    let s = seconds - m * 60;

    //format the result string with leadng zeros
    return `${h.toString().padStart(2, '0')}:${m.toString().padStart(2, '0')}:${s.toString().padStart(2, '0')}`
}

function humanReadable(seconds) {
    const divisors = [3600, 60, 1];

    let result = [];

    divisors.forEach(divs => {
        let calc = parseInt(seconds / divs);
        result.push(calc.toString().padStart(2, '0'));
        seconds = seconds - calc * divs;
    })

    return result.join(":");
}


console.log(humanReadable(0), '00:00:00', 'humanReadable(0)');
console.log(humanReadable(5), '00:00:05', 'humanReadable(5)');
console.log(humanReadable(60), '00:01:00', 'humanReadable(60)');
console.log(humanReadable(86399), '23:59:59', 'humanReadable(86399)');
console.log(humanReadable(359999), '99:59:59', 'humanReadable(359999)');

File: CodeWars/5kyu/Python/004-human-readable-time.py
# https://www.codewars.com/kata/52685f7382004e774f0001f7
# Write a function, which takes a non-negative integer (seconds) as input and returns the time in a human-readable format (HH:MM:SS)
# HH = hours, padded to 2 digits, range: 00 - 99
# MM = minutes, padded to 2 digits, range: 00 - 59
# SS = seconds, padded to 2 digits, range: 00 - 59
# The maximum time never exceeds 359999 (99:59:59)

def make_readable(seconds):
    hours = seconds // (60*60)
    seconds = seconds - hours*60*60
    minutes = seconds // 60
    seconds = seconds - minutes * 60

    return f'{str(hours).rjust(2,"0")}:{str(minutes).rjust(2,"0")}:{str(seconds).rjust(2,"0")}'


# solution from Codewars
def make_readable(seconds):
    return "{:0>2}:{:0>2}:{:0>2}".format(seconds // 3600, (seconds % 3600) // 60, (seconds % 3600) % 60)

print(make_readable(0), "00:00:00")
print(make_readable(5), "00:00:05")
print(make_readable(60), "00:01:00")
print(make_readable(86399), "23:59:59")
print(make_readable(359999), "99:59:59")